executor/src/witgen/jit/effect.rs

```diff
@@ -165,3 +176,96 @@ fn format_condition<T: FieldElement>(condition: &BranchCondition<T, Variable>) -
         Ordering::Greater => format!("{var} <= {min} || {var} >= {max}"),
     }
 }
+
+#[cfg(test)]
+mod test {
+    use powdr_number::GoldilocksField;
+
+    use crate::witgen::jit::variable::Cell;
+
+    use pretty_assertions::assert_eq;
+
+    use super::*;
+    type T = GoldilocksField;
+
+    fn var(id: u64) -> Variable {
+        Variable::WitnessCell(Cell {
+            column_name: format!("v{id}"),
+            id,
+            row_offset: 0,
+        })
+    }
+
+    #[test]
+    fn combine_if_else() {
+        let effects = vec![
+            Effect::Assignment(var(0), SymbolicExpression::from(T::from(1))),
+            Effect::Branch(
+                BranchCondition {
+                    variable: var(0),
+                    first_branch: RangeConstraint::from_range(T::from(1), T::from(2)),
+                    second_branch: RangeConstraint::from_range(T::from(3), T::from(4)),
+                },
+                vec![Effect::Assignment(
+                    var(1),
+                    SymbolicExpression::from(T::from(2)),
+                )],
+                vec![Effect::Branch(
+                    BranchCondition {
+                        variable: var(1),
+                        first_branch: RangeConstraint::from_range(T::from(5), T::from(6)),
+                        second_branch: RangeConstraint::from_range(T::from(7), T::from(8)),
```

Comment:
why does [7,8] not show in the generated code?

Reply:
The else case is implicit, it should be the negation of the first branch with respect to the current range constraints.

But let me check if we can simplify that.